### PR TITLE
PDF: deal with the situation where a multipage PDF has layers with same name but specific to some page(s)

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

File: frmts/pdf/pdfdataset.cpp [hunk 1]
```diff
@@ -4179,6 +4284,85 @@ PDFDataset::VisibilityState PDFDataset::GetVisibilityStateForOGCPdfium(int nNum,
...
+            const auto poXObject = poPage->LookupObject("Resources.XObject");
+            if (poXObject && poXObject->GetType() == PDFObjectType_Dictionary)
+            {
+                for (const auto &oNameObjectPair :
+                     poXObject->GetDictionary()->GetValues())
+                {
+                    const auto poProperties =
+                        oNameObjectPair.second->LookupObject(
+                            "Resources.Properties");
+                    if (poProperties != nullptr &&
```
Comment (elpaso) [inline frmts/pdf/pdfdataset.cpp:4332]:
```suggestion
                    if (poProperties &&
```

File: frmts/pdf/pdfdataset.cpp [hunk 2]
```diff
@@ -4179,6 +4284,85 @@ PDFDataset::VisibilityState PDFDataset::GetVisibilityStateForOGCPdfium(int nNum,
...
+                        for (const auto &[osKey, poObj] : oMap)
+                        {
+                            if (poObj->GetRefNum().toBool() &&
+                                poObj->GetType() == PDFObjectType_Dictionary)
+                            {
+                                GDALPDFObject *poType =
+                                    poObj->GetDictionary()->Get("Type");
+                                GDALPDFObject *poName =
+                                    poObj->GetDictionary()->Get("Name");
+                                if (poType != nullptr &&
```
Comment (elpaso) [inline frmts/pdf/pdfdataset.cpp:4346]:
```suggestion
                                if (poType &&
```

File: frmts/pdf/pdfdataset.cpp [hunk 3]
```diff
@@ -4179,6 +4284,85 @@ PDFDataset::VisibilityState PDFDataset::GetVisibilityStateForOGCPdfium(int nNum,
...
+                                    poType->GetType() == PDFObjectType_Name &&
+                                    poType->GetName() == "OCG" &&
+                                    poName != nullptr &&
```
Comment (elpaso) [inline frmts/pdf/pdfdataset.cpp:4349]:
```suggestion
                                    poName &&
```

**Review by elpaso — APPROVED** [review]

File: frmts/pdf/pdfdataset.cpp (hunk 1 above)
Comment (rouault) [reply frmts/pdf/pdfdataset.cpp]: done

File: frmts/pdf/pdfdataset.cpp (hunk 2 above)
Comment (rouault) [reply frmts/pdf/pdfdataset.cpp]: done

File: frmts/pdf/pdfdataset.cpp (hunk 3 above)
Comment (rouault) [reply frmts/pdf/pdfdataset.cpp]: done

**Pull request merged**